modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -3586,6 +3305,26 @@ private void checkNearCacheStarted(IgniteCacheProxy<?, ?> cache) throws IgniteCh
                 "(a cache with the same name without near cache is already started)");
     }
 
+    /**
+     * This method allows manually remove the checkpoint with given {@code key}.
+     *
+     * @param key Checkpoint key.
+     * @return {@code true} if specified checkpoint was indeed removed, {@code false}
+     *      otherwise.
+     */
+    boolean removeCheckpoint(String key) {
```

Comment:
I suggest leaving this method only in the bean.

Reply:
> I suggest leaving this method only in the bean.

@NSAmelchev, what do you suggest to do with IgniteKernal.checkClusterState() ?

Reply:
Ok, let's leave it here. And do not move it down in the code.